packages/core/plugins/core-plugins-server-internal/src/plugin_system_overrides.ts

```diff
@@ -0,0 +1,37 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+/**
+ * This class is used to override the default behavior of the plugin system.
+ *
+ * @note this is for internal use only!
+ *
+ * @internal
+ */
+export class PluginSystemOverrides {
```

Comment:
Side effect kills kittens. Lot of kittens. A hell lot of kittens.

We successfully avoided similar package-static patterns in our codebase until now. Can't we just use an undocumented configuration option instead of that thing? It would be significantly better ihmo.

Reply:
Sure, I was thinking of the config route, both have some nastiness, where this seemed to introduce the nastiness in, what I think of, as the simplest way.

But I take your point that if we've avoided this pattern explicitly no reason to introduce it now.